pkg/controllermanager/controller/shoot/shoot_control_reconcile.go

```diff
@@ -216,6 +216,16 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.ComputeShootOperatingSystemConfig).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(initializeShootClients, waitUntilInfrastructureReady),
 		})
+		deployNetworking = g.Add(flow.Task{
+			Name:         "Deploying Cluster Network Resources",
+			Fn:           flow.TaskFn(botanist.DeployNetwork).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(initializeShootClients, waitUntilInfrastructureReady, computeShootOSConfig),
```

Comment:
```suggestion
			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager),
```

Should be sufficient, or? Why should we wait for the OS config or infrastructure?

Reply:
lets also wait for infrastructure and OSConfig. With `usePodCIDR` for IPAM the value is picked from the `Node.podCIDR` and I guess nodes will not join without proper infra and OS config.